src/com/google/cose/CoseKey.java

```diff
@@ -95,4 +97,17 @@ public int getKeyType() {
   public java.util.Map<Integer, DataItem> getLabels() {
     return labels;
   }
+
+  void verifyOperationAllowedByKey(int keyOperation) throws CoseException {
+    if (operations != null && operations.contains(keyOperation)) {
```

Comment:
This check appears to be backwards. Shouldn't it be `!operations.contains(keyOperation)`?

Does this pass unit tests?

Reply:
this is checking if the operation is allowed by the key. So the operations list should contain the key operation.

All unit tests are passing.